### PR TITLE
MultiFab Fixture Cleanup via `FabArray::clear`

### PR DESCRIPTION
**Pull request by ax3l**
Using a context manager and calling clear ensures that we will not hold device memory anymore once we hit `AMReX::Finalize`, even in the situation where an exception is raised in a test. This avoids segfaults for failing tests.

Related to #81

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -85,47 +85,59 @@ def distmap(boxarr):
...
+            self.mfab.set_val(0.0, 0, num_components)
+            return self.mfab
 
-    return create
+        def __exit__(self, exc_type, exc_value, traceback):
+            self.mfab.clear()
+            del self.mfab
+
+    with MfabContextManager() as mfab:
+        yield mfab
```
Comment (sayerhs) [inline tests/conftest.py:109]: Could be simplified?

```python
def mfab(boxarr, distmap, request):
    num_components = request.param[0]
    num_ghost = request.param[1]
    try:
        mfab = amr.MultiFab(boxarr, distmap, num_components, num_ghost)
        mfab.set_val(0.0, 0, num_components)
        yield mfab
    finally:
        mfab.clear()
        del mfab

File: tests/conftest.py [hunk 2]
```diff
@@ -85,47 +85,59 @@ def distmap(boxarr):
...
-        num_components = request.param[0]
-        num_ghost = request.param[1]
-        mfab = amr.MultiFab(boxarr, distmap, num_components, num_ghost)
-        mfab.set_val(0.0, 0, num_components)
-        return mfab
+    class MfabContextManager:
+        def __enter__(self):
+            num_components = request.param[0]
+            num_ghost = request.param[1]
+            self.mfab = amr.MultiFab(boxarr, distmap, num_components, num_ghost)
```
Comment (sayerhs) [inline tests/conftest.py:100]: `pylint` would probably create warnings for this.

File: tests/conftest.py (hunk 1 above)
Comment (ax3l) [reply tests/conftest.py:109]: According to my knowledge, that will cache the MultiFab between runs and break out memory pool assumptions...

Worth to try again

File: tests/conftest.py (hunk 2 above)
Comment (ax3l) [reply tests/conftest.py:100]: Oh really? Linting still coming in #208 😅

CodeQL did not yet complain, what's to be avoided here?

File: tests/conftest.py (hunk 1 above)
Comment (sayerhs) [reply tests/conftest.py:109]: Interesting. The fixture is marked function scope, so it shouldn't cache a value. Is it pytest that's caching the value?

File: tests/conftest.py (hunk 2 above)
Comment (sayerhs) [reply tests/conftest.py:100]: Pylint will complain about attributes being declared/created in a non-init function.

File: tests/conftest.py (hunk 1 above)
Comment (ax3l) [reply tests/conftest.py:109]: I thought so... maybe I misunderstood something when caching kicks in.

**Comment by ax3l** [comment]
@sayerhs I wonder if I shall just merge this for now and refactor it later, since it fixes issues we see on Perlmutter...?

**Pull request merged**